cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -360,6 +361,15 @@ class SnappyUnifiedMemoryManager private[memory](
           return
         }
 
+        // Stop execution pool to grow beyond eviction percentage of heap.
+        if (memoryMode eq MemoryMode.ON_HEAP) {
+          if (executionPool.memoryUsed + extraMemoryNeeded > maxHeapExecutionSize) {
+            logWarning(s"MemoryManager can't allocate $numBytes bytes as it " +
+                s"would exceed maxHeapExecutionSize = $maxHeapExecutionSize")
+            return
+          }
+        }
+
```

Comment:
Not clear what this is accomplishing and why only for heap. What was the issue with previous way which would grow execution till storageFraction (unless there was some bug in that)?